app/controllers/admin/videos_controller.rb

```diff
@@ -0,0 +1,24 @@
+class Admin::VideosController < AdminController
+  def new 
+    @video = Video.new
+    @category_options = Category.all.map { |c| [c.name, c.id] }
+  end
+
+  def create
+    @video = Video.new(video_params)
+
+    if @video.save
+      flash[:notice] = "Your successfully added #{@video.title}"
+      redirect_to new_admin_video_path
+    else
+      flash[:error] = "There was an error, please try again"
```

Comment:
When you render, it should be `flash.now` otherwise you are going to pop the flash on next GET request.